### PR TITLE
use `run_shell_cmd` in custom easyblock for Mathematica

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyblocks/m/mathematica.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ##
-# Copyright 2013-2024 Ghent University
+# Copyright 2013-2023 Ghent University
```
Comment (Micket) [inline easybuild/easyblocks/m/mathematica.py:2]:
```suggestion
# Copyright 2013-2024 Ghent University
```

**Review by boegel — APPROVED** [review]
lgtm

File: easybuild/easyblocks/m/mathematica.py [hunk 2]
```diff
@@ -76,16 +76,17 @@ def install_step(self):
...
-                r"Create directory (y/n)? >": 'y',
-                r"Should the installer attempt to make this change (y/n)? >": 'n',
-                r"or press ENTER to select /usr/local/bin: >": os.path.join(self.installdir, "bin"),
-            }
+            qa = [
+                (r"Enter the installation directory, or press ENTER to select %s: >" % qa_install_path,
+                 self.installdir),
+                (r"Create directory \(y/n\)\? >", 'y'),
+                (r"Should the installer attempt to make this change \(y/n\)\? >", 'n'),
+                (r"or press ENTER to select /usr/local/bin: >", os.path.join(self.installdir, "bin")),
```
Comment (boegel) [inline easybuild/easyblocks/m/mathematica.py:84]: There are a couple of line breaks in this question, so we need to use `[\s\n]*`:
```suggestion
               (r"or press ENTER to select[\s\n]*/usr/local/.*:[\s\n]*>", os.path.join(self.installdir, "bin")),
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/m/mathematica.py [hunk 3]
```diff
@@ -76,16 +76,17 @@ def install_step(self):
...
-                r"Create directory (y/n)? >": 'y',
-                r"Should the installer attempt to make this change (y/n)? >": 'n',
-                r"or press ENTER to select /usr/local/bin: >": os.path.join(self.installdir, "bin"),
-            }
+            qa = [
+                (r"Enter the installation directory, or press ENTER to select %s: >" % qa_install_path,
+                 self.installdir),
+                (r"Create directory \(y/n\)\? >", 'y'),
+                (r"Should the installer attempt to make this change \(y/n\)\? >", 'n'),
+               (r"or press ENTER to select[\s\n]*/usr/local/.*:[\s\n]*>", os.path.join(self.installdir, "bin")),
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyblocks/m/mathematica.py:84]:
```suggestion
                (r"or press ENTER to select[\s\n]*/usr/local/.*:[\s\n]*>", os.path.join(self.installdir, "bin")),

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]

File: easybuild/easyblocks/m/mathematica.py [hunk 4]
```diff
@@ -76,16 +76,17 @@ def install_step(self):
...
-            qa = {
-                r"Enter the installation directory, or press ENTER to select %s: >" % qa_install_path: self.installdir,
-                r"Create directory (y/n)? >": 'y',
-                r"Should the installer attempt to make this change (y/n)? >": 'n',
-                r"or press ENTER to select /usr/local/bin: >": os.path.join(self.installdir, "bin"),
-            }
+            qa = [
+                (r"Enter the installation directory, or press ENTER to select %s: >" % qa_install_path,
+                 self.installdir),
+                (r"Create directory \(y/n\)\? >", 'y'),
```
Comment (boegel) [inline easybuild/easyblocks/m/mathematica.py:82]:
```suggestion
                (r"Create directory \(y/n\)\?[\s\n]*>", 'y'),
```

File: easybuild/easyblocks/m/mathematica.py [hunk 5]
```diff
@@ -76,16 +76,17 @@ def install_step(self):
...
+                (r"Should the installer attempt to make this change \(y/n\)\? >", 'n'),
```
Comment (boegel) [inline easybuild/easyblocks/m/mathematica.py:83]:
```suggestion
                (r"Should the installer attempt to make this change \(y/n\)\?[\s\n]*>", 'n'),
```

File: easybuild/easyblocks/m/mathematica.py [hunk 6]
```diff
@@ -76,16 +76,17 @@ def install_step(self):
...
             shortver = '.'.join(self.version.split('.')[:2])
             qa_install_path = os.path.join('/usr', 'local', 'Wolfram', self.name, shortver)
-            qa = {
-                r"Enter the installation directory, or press ENTER to select %s: >" % qa_install_path: self.installdir,
-                r"Create directory (y/n)? >": 'y',
-                r"Should the installer attempt to make this change (y/n)? >": 'n',
-                r"or press ENTER to select /usr/local/bin: >": os.path.join(self.installdir, "bin"),
-            }
+            qa = [
+                (r"Enter the installation directory, or press ENTER to select %s: >" % qa_install_path,
```
Comment (boegel) [inline easybuild/easyblocks/m/mathematica.py:80]:
```suggestion
                (r"Enter the installation directory, or press ENTER to select %s:[\s\n]*>" % qa_install_path,
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/m/mathematica.py [hunk 7]
```diff
@@ -76,16 +76,17 @@ def install_step(self):
...
-                r"Create directory (y/n)? >": 'y',
-                r"Should the installer attempt to make this change (y/n)? >": 'n',
-                r"or press ENTER to select /usr/local/bin: >": os.path.join(self.installdir, "bin"),
-            }
+            qa = [
+                (r"Enter the installation directory, or press ENTER to select %s:[\s\n]*>" % qa_install_path,
+                 self.installdir),
+                (r"Create directory \(y/n\)\?[\s\n]*>", 'y'),
+                (r"Should the installer attempt to make this change \(y/n\)\?[\s\n]*>", 'n'),
+                (r"or press ENTER to select[\s\n]*/usr/local/.*:[\s\n]*>", os.path.join(self.installdir, "bin")),
```
Comment (boegel) [inline easybuild/easyblocks/m/mathematica.py:84]:
```suggestion
                (r"or press ENTER to select[\s\n]*/usr/local/bin:[\s\n]*>", os.path.join(self.installdir, "bin")),
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/m/mathematica.py [hunk 8]
```diff
@@ -76,16 +76,17 @@ def install_step(self):
...
             shortver = '.'.join(self.version.split('.')[:2])
             qa_install_path = os.path.join('/usr', 'local', 'Wolfram', self.name, shortver)
-            qa = {
-                r"Enter the installation directory, or press ENTER to select %s: >" % qa_install_path: self.installdir,
-                r"Create directory (y/n)? >": 'y',
-                r"Should the installer attempt to make this change (y/n)? >": 'n',
-                r"or press ENTER to select /usr/local/bin: >": os.path.join(self.installdir, "bin"),
-            }
+            qa = [
+                (r"Enter the installation directory, or press ENTER to select %s:[\s\n]*>" % qa_install_path,
```
Comment (boegel) [inline easybuild/easyblocks/m/mathematica.py:80]:
```suggestion
                (r"Enter the installation directory, or press ENTER to select[\s\n]*%s:[\s\n]*>" % qa_install_path,

**Review by boegel — CHANGES_REQUESTED** [review]
fix question pattern to specify installation prefix in Mathematica easyblock

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Mathematica-12.1.1.eb_ 
 * **SUCCESS** _Mathematica-13.1.0.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3564.doduo.os - Linux RHEL 8.8, x86_64, AMD EPYC 7552 48-Core Processor, Python 3.6.8
See https://gist.github.com/boegel/91aa51c042a01ff063e2f9b3291f207a for a full test report.

**Pull request merged**